src/AggregateFunctions/AggregateFunctionMap.h

```diff
@@ -187,6 +212,33 @@ class AggregateFunctionMap final
         }
     }
 
+    template <bool up_to_state>
+    void destroyImpl(AggregateDataPtr __restrict place) const noexcept
+    {
+        AggregateFunctionMapCombinatorData<KeyType> & state = Base::data(place);
+
+        for (const auto & [key, nested_place] : state.merged_maps)
+        {
+            if constexpr (up_to_state)
+                nested_func->destroyUpToState(nested_place);
+            else
+                nested_func->destroy(nested_place);
+        }
+
+        auto to_destroy = std::move(state.merged_maps);
```

Comment:
Why is it moved? It'll be destroyed on the next line.

Reply:
Forgot to remove it after I added next line, fixed